pkg/log/zap.go

```diff
@@ -0,0 +1,80 @@
+package log
+
+import (
+	"os"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// Config is the configuration for the logger.
+type Config struct {
+	// LogLevel is the log level to use. The default is "info".
+	LogLevel string
+	// WriteTo is the path to write logs to. The default is stderr.
+	WriteTo string
+}
+
+// NewLogger creates a new logger with the given configuration. This logger
+// is pre-configured to production settings. To change the settings, modify
+// the Config struct.
+func NewLogger(config Config) *zap.Logger {
+	// EncodeTime is set to ISO8601TimeEncoder by default. This is a more human readable
+	// format than the default EpochTimeEncoder.
+	encoderCfg := zap.NewProductionEncoderConfig()
+	encoderCfg.EncodeTime = zapcore.ISO8601TimeEncoder
+
+	// outputPaths is the list of paths to write logs to. By default, logs are
+	// written to stderr.
+	outputPaths := []string{"stderr"}
+	if config.WriteTo != "" {
+		outputPaths = append(outputPaths, config.WriteTo)
+	}
+
+	// initialFields are the fields that are added to every log message.
+	initialFields := map[string]interface{}{
+		"pid": os.Getpid(),
+	}
+	if config.WriteTo != "" {
+		initialFields["writing_logs_to"] = config.WriteTo
```

Comment:
This means every log will have a property like the following:
`"writing_logs_to": "<the file you're currently looking at>"` right?

Reply:
removed it so it is not part of every log